api/models/group.py

```diff
@@ -122,11 +122,15 @@ def __str__(self) -> str:
             f"teachers: {teacher_names}, {self.students.count()} students."
         )
 
-    def teachers_with_other_groups(self) -> QuerySet[Teacher]:
-        return self.teachers.annotate(group_count=Count("groups")).filter(group_count__gt=1)
+    def _active_group_count_annotation(self) -> QuerySet[Teacher]:
+        active_filter = Q(groups__project_status__in=GroupProjectStatus.active_statuses())
```

Comment:
Please, pay attention to this, I realized that we should count only `active` groups because teacher might have tons of already finished groups.

Reply:
Hmm no actually I think they won't: the former groups will be in a teacher's `groups_former` attribute (see `related_name`):

```py
    teachers_former = models.ManyToManyField(
        Teacher,
        blank=True,
        related_name="%(class)ss_former",
        verbose_name="Former teachers",
        help_text=(
            "Lists teachers that once worked with this group, and/or the teacher(s) of this group "
            "when it finished classes or was aborted."
        ),
    )
```

Reply:
Hmm, yes, then it does not change anything, I would still keep it as that to insure that invariant but on the contrary it might confuse people in some point, not sure

Reply:
I would suggest removing it to avoid confusion. No former group must remain in teacher's `.groups()`, and applying a filter might mask a coding mistake.